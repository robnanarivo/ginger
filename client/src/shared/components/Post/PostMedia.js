import { React } from 'react';
import CardMedia from '@mui/material/CardMedia';
import ReactAudioPlayer from 'react-audio-player';
import ReactPlayer from 'react-player';

const PostMedia = ({ content }) => {
  if (content.image) {
    return (
      <CardMedia
        component="img"
        height="194"
        image={content.image}
        alt="postimage"
      />
    );
  }
  if (content.audio) {
    return (
      <ReactAudioPlayer
        src={content.audio}
        controls
      />
    );
  }
  if (content.video) {
    return (
      <ReactPlayer url={content.video} width="550px" controls />
    );
  }
  return '';
};

export default PostMedia;
