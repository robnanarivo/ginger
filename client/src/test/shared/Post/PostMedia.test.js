import React from 'react';
import renderer from 'react-test-renderer';
import PostMedia from '../../../shared/components/Post/PostMedia';
import 'jest-styled-components';

describe('PostMedia unit test', () => {
  it('PostMedia Snapshot test', () => {
    const tree = renderer
      .create(
        <PostMedia content={{ text: '123' }} />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
