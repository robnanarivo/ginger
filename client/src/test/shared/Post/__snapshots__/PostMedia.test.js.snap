// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostMedia unit test PostMedia Snapshot test 1`] = `""`;
